fix(search-page): modified layout to add spacing tweaks to fit the new design

jimmypalelil opened the pull request:
# Description

Added spacing tweaks and layout modifications in the **Search** page based on the design in this ticket: #489 


# Screenshots (search page)
<img width="2222" alt="image" src="https://github.com/user-attachments/assets/fe4eff6c-dd66-4333-82eb-0f3a11714cb5" />
<img width="2010" alt="image" src="https://github.com/user-attachments/assets/0986e0ee-e1b7-439a-a17a-13c5170b46ae" />

marcellmueller commented:
Nice work, this looks mostly good though I wonder if we should sync with Harry or Jamie about the spacing on the search page, the gap looks really massive in practice especially with the longer filter names hidden behind the `show more` menu 🤔 

Another thing I noticed is an alignment issue on the recreation resource page:
<img width="663" alt="Screenshot 2025-04-09 at 11 48 58 AM" src="https://github.com/user-attachments/assets/b7b3daf4-5b8e-43ef-bfa6-af05c2ba6a16" />

Figma:
<img width="663" alt="Screenshot 2025-04-09 at 11 49 27 AM" src="https://github.com/user-attachments/assets/8f35c4c9-54c2-440a-9601-b0902fc827e7" />

jimmypalelil commented:
### Fixes

- Fixed the gap in the search page between filters and results
- fixed the alignment between header, nav, and page contents in both search and rec resource pages
 
---
### Screenshots (multiple screen sizes)

[rec resource page.zip](https://github.com/user-attachments/files/19676348/rec.resource.page.zip)
[search page.zip](https://github.com/user-attachments/files/19676349/search.page.zip)

marcellmueller commented on frontend/src/styles/variables.scss:
```diff
@@ -29,7 +29,6 @@ $mdBreakpoint: 768px;
-$pagePaddingLg: 16px;
+$pagePadding: 2rem;
+$pagePaddingLg: 1rem;
+$pageMaxWidth: calc(1080px - ($pagePaddingLg * 2));
```
In a vacuum this is a good way to set the width though bcparks.ca uses a fixed 1060px width. 

I think this calculation is off and causes our width to be larger than bcparks, I noticed in the Happo diffs that the pages are a bit wider now. I compared to bcparks.ca visually and our page is a bit wider than theirs now, also confirmed in dev tools:

<img width="1470" alt="Screenshot 2025-04-10 at 8 21 27 AM" src="https://github.com/user-attachments/assets/66de06d7-355a-4c7e-9f9b-91503c5cdc7e" />
<img width="1470" alt="Screenshot 2025-04-10 at 8 24 41 AM" src="https://github.com/user-attachments/assets/0b4ea6df-c90b-4b9f-a88d-3b286e1d6aff" />

marcellmueller reviewed (approved):
Thanks and sorry about the confusion with the page width 🙇
